utils/s2n_compiler.h

```diff
@@ -22,3 +22,11 @@
 #define S2N_GCC_VERSION_AT_LEAST(major, minor, patch_level) \
     ((S2N_GCC_VERSION) >= ((major) * 10000 + (minor) * 100 + (patch_level)))
 
+#if defined (__clang__)
+    #define S2N_GCC_VERSION_AT_LEAST_OR_CLANG(major, minor, patch_level) \
```

Comment:
This new macro shouldn't be necessary. Why not just use:
```
#if defined (__clang__) || S2N_GCC_VERSION_AT_LEAST(x, y, z)
```
Otherwise, it seems like we'd end up defining a lot of variations on S2N_GCC_VERSION_AT_LEAST :)